src/fs/src/lib.rs

```diff
@@ -73,8 +70,8 @@ bitflags! {
 }
 
 impl From<u16> for FileMode {
-    fn from(value: u16) -> Self {
-        Self { bits: value }
+    fn from(item: u16) -> Self {
+        Self::from_bits_truncate(item)
```

Comment:
This will silently drop any bits that were not defined. Better to use `from_bits_retain` instead.